### PR TITLE
[LED-100] Added Exporting of data

### PR DESCRIPTION
**Pull request by shellajt**
Turned out to be a little more complicated than we though. Modified the DB controller to store a reference to the storage location. I had to shutdown the db before copying the file (darn file locks). After the copy I prompt the user for the password again and reconnect them to the db

### CODE REVIEW

**Comment by LadyMopar** [comment]
Not necessary, but the confirm password popup might look better if the prompt text was all centered instead of left-aligned in the center. Also, instead of saying "Login" again, maybe say "Continue". I feel like that's a little more intuitive to the user. Otherwise, it works great!

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 1]
```diff
@@ -141,4 +153,46 @@ private void createAccountPopup() {
         Scene scene = new Scene(accountController);
         this.createModal(scene, "Add Account");
     }
+
+    /**
+     * Exports the database file to the chosen directory.
+     */
+    private void exportData() {
+        JFileChooser chooser = new JFileChooser();
```
Comment (LadyMopar) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Why are we using Swing components here? Wouldn't it be better to stay consistent with the JavaFX file chooser? The Swing one works sloppily and looks outdated...

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 2]
```diff
@@ -141,4 +153,46 @@ private void createAccountPopup() {
...
+            this.setupErrorPopup("Unable to properly close database.", e);
+            e.printStackTrace();
+        }
+        this.displayPasswordPrompt();
+    }
+
+    private void displayPasswordPrompt() {
+        PasswordPromptController promptController = new PasswordPromptController();
+        Scene scene = new Scene(promptController);
+        this.createModal(scene, "Login");
```
Comment (LadyMopar) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Maybe say "Continue" or "Verify Password" instead of "Login"

Comment (shellajt) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Changed to "Continue"

**Pull request merged**